src/drivers/ulanding/ulanding.cpp

```diff
@@ -0,0 +1,392 @@
+
+		/* do regular cdev init */
+		ret = CDev::init();
+
+		if (ret != OK) {
+			PX4_WARN("cdev init failed");
+			break;
+		}
+
+		int fd = px4_open(RANGE_FINDER0_DEVICE_PATH, 0);
+
+		if (fd < 0) {
+			PX4_WARN("failed to open range finder device");
+			ret = 1;
+			break;
+		}
+
+		px4_close(fd);
+
+		/* open fd */
+		fd = px4_open(_port, O_RDWR | O_NOCTTY);
+
+		if (fd < 0) {
+			PX4_WARN("failed to open serial device");
+			ret = 1;
+			break;
+		}
+
+		struct termios uart_config;
+
+		int termios_state;
+
+		/* fill the struct for the new configuration */
+		tcgetattr(fd, &uart_config);
+
+		/* clear ONLCR flag (which appends a CR for every LF) */
+		uart_config.c_oflag &= ~ONLCR;
+
+		/* no parity, one stop bit */
+		uart_config.c_cflag	 &= ~(CSTOPB | PARENB);
+
+		unsigned speed = B115200;
+
+		/* set baud rate */
+		if ((termios_state = cfsetispeed(&uart_config, speed)) < 0) {
+			PX4_WARN("ERR CFG: %d ISPD", termios_state);
+			ret = 1;
+			break;
+		}
+
+		if ((termios_state = cfsetospeed(&uart_config, speed)) < 0) {
+			PX4_WARN("ERR CFG: %d OSPD\n", termios_state);
+			ret = 1;
+			break;
+		}
+
+		if ((termios_state = tcsetattr(fd, TCSANOW, &uart_config)) < 0) {
+			PX4_WARN("ERR baud %d ATTR", termios_state);
+			ret = 1;
+			break;
+		}
+
+		px4_close(fd);
+
+		_class_instance = register_class_devname(RANGE_FINDER_BASE_DEVICE_PATH);
+
+		struct distance_sensor_s ds_report = {};
+
+		_distance_sensor_topic = orb_advertise_multi(ORB_ID(distance_sensor), &ds_report,
+					 &_orb_class_instance, ORB_PRIO_HIGH);
+
+		if (_distance_sensor_topic == nullptr) {
+			DEVICE_LOG("failed to create distance_sensor object. Did you start uOrb?");
+			ret = 1;
+			break;
+		}
+
+	} while (0);
+
+	return ret;
+}
+
+void
+Radar::task_main_trampoline(int argc, char *argv[])
+{
+	radar::g_dev->task_main();
+}
+
+int
+Radar::start()
+{
+	ASSERT(_task_handle == -1);
+
+	/* start the task */
+	_task_handle = px4_task_spawn_cmd("radar",
+					  SCHED_DEFAULT,
+					  SCHED_PRIORITY_MAX - 30,
+					  800,
+					  (px4_main_t)&Radar::task_main_trampoline,
+					  nullptr);
+
+	if (_task_handle < 0) {
+		PX4_WARN("task start failed");
+		return -errno;
+	}
+
+	return OK;
+}
+
+void
+Radar::task_main()
+{
+
+	int fd = px4_open(_port, O_RDWR | O_NOCTTY);
+
+	// we poll on data from the serial port
+	px4_pollfd_struct_t fds[1];
+	fds[0].fd = fd;
+	fds[0].events = POLLIN;
+
+	// read buffer, one measurement consists of three bytes
+	char buf[6];
+
+	while (!_task_should_exit) {
+		// wait for up to 100ms for data
+		int pret = px4_poll(&fds[0], (sizeof(fds) / sizeof(fds[0])), 100);
+
+		// timed out
+		if (pret == 0) {
+			continue;
+		}
+
+		if (pret < 0) {
+			PX4_DEBUG("radar serial port poll error");
+			// sleep a bit before next try
+			usleep(100000);
+			continue;
+		}
+
+		if (fds[0].revents & POLLIN) {
+
+			int bytesAvailable = 0;
+			px4_ioctl(fd, FIONREAD, (unsigned long)&bytesAvailable);
```

Comment:
@LorenzMeier I actually put that in after because I assumed the problem could be that not enough bytes are read. It does not make a difference, I tried lots of stuff!